core/Common.php

```diff
@@ -574,6 +575,11 @@ public static function getRequestVar($varName, $varDefault = null, $varType = nu
             settype($value, $varType);
         }
 
+        if (is_array($requestArrayToUse[$varName])) {
+            $value = implode(",", $value);
+        }
```

Comment:
I don't think that is a proper solution, as it might break other places, where we expect an array, but don't pass a type maybe.

Reply:
@sgiehl right, I guess another solution is to do a name check, but that looks urgly. Or we update everyplace has common:getRequestVar('period') and common:getRequestVar('date')
```php
 if (is_array($requestArrayToUse[$varName]) && in_array($varName,['period', 'date']) ) {
            $value = implode(",", $value);
        }
```

Reply:
We should avoid such special handling I would say. Personally I would prefer a method like `Piwik::getDate($default=null)` / `Piwik::getPeriod($default=null)`, that performs the `getRequestVar` correctly and use that everywhere. Like we have `Piwik::getModule()`
But maybe let's discuss that on #18639